packages/react-button/src/components/ToggleButton/ToggleButton.types.ts

```diff
@@ -1,33 +1,24 @@
-import type { ComponentStateCompat } from '@fluentui/react-utilities';
-import type {
-  ButtonDefaultedProps,
-  ButtonProps,
-  ButtonShorthandPropsCompat,
-  ButtonState,
-} from '../Button/Button.types';
+import type { ButtonProps, ButtonState } from '../Button/Button.types';
 
-export interface ToggleButtonProps extends ButtonProps {
+export interface ToggleButtonCommons {
   /**
    * Defines the controlled checked state of the `ToggleButton`.
    * Mutually exclusive to `defaultChecked`.
    * This should only be used if the checked state is to be controlled at a higher level and there is a plan to pass the
    * correct value based on handling `onClick` events and re-rendering.
    * @default false
    */
-  checked?: boolean;
-
-  /**
-   * Defines whether the `ToggleButton` is initially in a checked state or not when rendered.
-   * Mutually exclusive to `checked`.
-   * @default false
-   */
-  defaultChecked?: boolean;
+  checked: boolean;
 }
 
-export type ToggleButtonShorthandPropsCompat = ButtonShorthandPropsCompat;
-
-export type ToggleButtonDefaultedProps = ButtonDefaultedProps;
+export type ToggleButtonProps = ButtonProps &
+  Partial<ToggleButtonCommons> & {
+    /**
+     * Defines whether the `ToggleButton` is initially in a checked state or not when rendered.
+     * Mutually exclusive to `checked`.
+     * @default false
+     */
+    defaultChecked?: boolean;
+  };
```

Comment:
This type is not correct. It doesn't ensure the mutual exclusivity of `checked` and `defaultChecked` and excludes the latter from `ToggleButtonCommons`. I believe it should be something like:

```ts
export type ToggleButtonCommons = 
  ({ checked: boolean; } | { defaultChecked: boolean; })
  & { /* optional shared props */ }

// ...

export type ToggleButtonProps = ButtonProps & Partial<ToggleButtonCommons>
```

Reply:
Not correct is too strong IMO.

they're not mutually exclusive although one overrides the other, it's pretty much the same approach for `value` and `defaultValue` on `input` for example, and there's no need for mutual exclusivity in `input` case and we should follow react way of declaring it.

Reply:
@bsunderhus "Not correct" was based on comments describing both properties. If both properties say that the property is mutually exclusive to each other then the type that allows so is not correct. If we do however want a different behaviour as the comments describe then the comments should be updated accordingly.

Reply:
I've updated the comments to remove that part.